### ISSUE TITLE
round method works incorrectly for zero decimal places

### ISSUE THREAD

**Issue by ice1080** [akubera/bigdecimal-rs#136]
The `.round(round_digits)` method does not work for rounding to zero decimal places.

```rust
let unrounded = BigDecimal::from_str("-0.68")?;
println!("{}", unrounded.round(0));
// output is 1 -- incorrect
println!("{}", unrounded.with_scale_round(0, RoundingMode::HalfUp));
// output is -1 -- correct
```

I do recognize that the docs for `round` specify that it is used for rounding to round_digits _after_ the decimal point, but it seems confusing that a round method doesn't work for round_digits less than 1.

> Return number rounded to round_digits precision after the decimal point

Shouldn't `round` work the same as `with_scale_round` just using the default rounding mode?

**Comment by danilopedraza** [comment]
I think is because of this [line](https://github.com/akubera/bigdecimal-rs/blob/37e7f13dad2c27b60f5b5f14e476d37bee3cdb4c/src/lib.rs#L813):

```Rust
return BigDecimal::new(BigInt::one(), round_digits);
```

It doesn't take the sign into account.

**Comment by danilopedraza** [comment]
Put a fix in #137

**Comment by akubera** [comment]
Fix has been merged in #137. 

Your solution was correct: `round` has been reimplemented as `with_scale_round(scale, <DEFAULT-ROUNDING-MODE>)`

On the subject of rounding, I have a commit here with new methods for Context and BigDecimalRef, if anyone has ideas for any particularly convenient methods which should be added, let me know:

https://github.com/akubera/bigdecimal-rs/commit/d7e8691e6d53ad9de78f83a87dff76914c4d093c
